bftengine/src/preprocessor/RequestProcessingState.hpp

```diff
@@ -146,6 +149,7 @@ class RequestProcessingState {
   // Maps result hash to a list of replica signatures sent for this hash. This also implicitly gives the number of
   // replicas returning a specific hash.
   std::map<concord::util::SHA3_256::Digest, std::list<PreProcessResultSignature>> preProcessingResultHashes_;
+  std::map<concord::util::SHA3_256::Digest, std::unordered_set<bftEngine::impl::NodeIdType>> seenSenders_;
```

Comment:
I've checked the serialization/deserialization mechanism in PreProcessResultSignature. It is not affected by the type of used container. I think it is much cleaner to replace the list with the set here. It will also avoid unneeded checks in PreProcessResultMsg::validatePreProcessResultSignatures function.

Reply:
yep, the API ```std::string serializeResultSignatureList(const std::list<PreProcessResultSignature>& sigs)``` confused me, I'll use a set and convert it to a list when calling RequestProcessingState::getPreProcessResultSignatures()

Reply:
I would change the container type end-to-end.